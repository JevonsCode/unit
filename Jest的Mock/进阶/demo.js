// 不会调用这里的 mock

// import axios from "axios";

// export const fetchData = () => {
//     return axios.get("/").then(res => res.data);
// }

export const getNumber = () => {
    return 100
}