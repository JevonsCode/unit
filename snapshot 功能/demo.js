export const generateConfig = () => {
    return {
        server: "http://0.0.0.0",
        port: 8081,
        domain: "local",
        time: new Date()
    }
}

export const generateConfig2 = () => {
    return {
        server: "http://0.0.0.0",
        port: 8082,
        domain: "local",
        time: new Date()
    }
}