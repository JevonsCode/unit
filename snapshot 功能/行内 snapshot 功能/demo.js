export const generateConfig = () => {
    return {
        server: "http://0.0.0.0",
        port: 8083,
        domain: "local",
        time: new Date()
    }
}