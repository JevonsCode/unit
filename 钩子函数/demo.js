export default class demo {
    constructor() {
        this.count = 0;
    }

    add () {
        this.count += 1;
    }

    minus () {
        this.count -= 1;
    }
}