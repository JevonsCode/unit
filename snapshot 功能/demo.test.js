import { generateConfig, generateConfig2 } from "./demo";

test("测试", () => {
    expect(generateConfig()).toMatchSnapshot({
        time: expect.any(Date)
    });
});

test("测试", () => {
    expect(generateConfig2()).toMatchSnapshot({
        time: expect.any(Date)
    });
});