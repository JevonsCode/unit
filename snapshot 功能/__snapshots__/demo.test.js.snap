// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 1`] = `
Object {
  "domain": "local",
  "port": 8081,
  "server": "http://0.0.0.0",
  "time": Any<Date>,
}
`;

exports[`测试 2`] = `
Object {
  "domain": "local",
  "port": 8082,
  "server": "http://0.0.0.0",
  "time": Any<Date>,
}
`;
